Wikipedia/Code/WMFSearchViewController.m

```diff
@@ -392,20 +392,28 @@ - (BOOL)textFieldShouldClear:(UITextField*)textField {
 #pragma mark - Search
 
 - (MWKSite*)currentlySelectedSearchSite{
-    NSUInteger index = [self.languageButtons indexOfObjectPassingTest:^BOOL(UIButton*  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
-        if(obj.selected){
-            *stop = YES;
-            return YES;
+    if([[NSUserDefaults standardUserDefaults] wmf_showSearchLanguageBar]){
+        
+        NSUInteger index = [self.languageButtons indexOfObjectPassingTest:^BOOL(UIButton*  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
disregard `NSIntegerMax == LONG_MAX` which is less than `ULONG_MAX`, so it's safe to cast to `NSUInteger` :ok_hand: